scripts/west_commands/runners/jlink.py

```diff
@@ -138,8 +108,34 @@ def dev_id_help(cls) -> str:
     def tool_opt_help(cls) -> str:
         return "Additional options for JLink Commander, e.g. '-autoconnect 1'"
 
+    @staticmethod
+    def default_jlink():
+        global DEFAULT_JLINK_EXE
+
+        if sys.platform == 'win32':
+            # JLink.exe can collide with the JDK executable of the same name
+            # Locate the executable using the registry
+            try:
+                import winreg
+                # Note that when multiple JLink versions are installed on the
+                # machine this seems to point to the one that was installed
+                # last, and not to the latest version.
+                key = winreg.OpenKeyEx(winreg.HKEY_LOCAL_MACHINE,
```

Comment:
Does this key exist if the SEGGER tools are only installed for the local user?

Reply:
You are very right. Thankfully, `HKEY_CURRENT_USER` seems to contain the info regardless of which installation scope was chosen, so I switched to that.